File: web/src/types/document.ts
export type DocType =
  | 'requirements'
  | 'design'
  | 'testing'
  | 'meeting_minutes'
  | 'management'
  | 'configuration'
  | 'general';
export type DocumentStatus = 'draft' | 'review' | 'submitted' | 'archived';

export interface DocumentGenerateRequest {
  docType: DocType;
  title: string;
  context: Record<string, unknown>;
  template?: string;
}

export interface DocumentTaskResponse {
  docTaskId: string;
  status: 'pending' | 'running' | 'completed' | 'failed' | 'cancelled';
  createdAt: string;
  resultDocId?: string;
  errorMessage?: string;
}

export interface DocumentSummary {
  docId: string;
  title: string;
  type: DocType;
  status: DocumentStatus;
  version: string;
  createdAt: string;
  submittedAt?: string;
  generatedBy?: string;
  fileSize?: number;
}

export interface DocumentSection {
  id: string;
  title: string;
  status: 'generated' | 'pending' | 'edited';
}

export interface DocumentDetail extends DocumentSummary {
  content: string;
  metadata: {
    createdAt: string;
    generatedBy: string;
    sourceTasks: string[];
    modelUsed: string;
  };
  sections: DocumentSection[];
}

export interface DocumentVersion {
  version: string;
  changeDescription?: string;
  createdAt: string;
  createdBy?: string;
}

// ----------------------------------------------------- 对话式文档生成

export type ReasoningStepStatus = 'pending' | 'running' | 'completed' | 'failed';

export interface ReasoningStep {
  id: string;
  title: string;
  detail?: string;
  status: ReasoningStepStatus;
}

export interface ReasoningSummary {
  status: 'thinking' | 'completed' | 'failed';
  steps: ReasoningStep[];
}

export interface DocumentConversationSummary {
  conversationId: string;
  title: string;
  docType?: DocType | null;
  documentId?: string | null;
  mode: 'doc_chat' | 'qa';
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface DocumentMessage {
  messageId: string;
  role: 'user' | 'assistant';
  content: string;
  docVersion?: string | null;
  reasoningSummary?: ReasoningSummary | null;
  createdAt: string;
}

export interface ConversationDetail extends DocumentConversationSummary {
  messages: DocumentMessage[];
  document: DocumentDetail | null;
}

export interface SendMessageResponse {
  conversationId: string;
  assistantMessage: DocumentMessage;
  document: DocumentDetail;
}

export interface DocumentSubmitResponse {
  docId: string;
  status: DocumentStatus;
  submittedAt: string;
  submissionRecord: {
    submitter: string;
    version: string;
    filePath: string;
    checksum: string;
  };
}

// ----------------------------------------------------- Q&A 模式
export interface QaSendResponse {
  conversationId: string;
  assistantMessage: DocumentMessage;
}

export type QaStreamEvent =
  | { type: 'reasoning'; summary: ReasoningSummary }
  | { type: 'answer'; assistantMessage: DocumentMessage }
  | { type: 'done' }
  | { type: 'error'; message: string };

File: server/app/schemas/document.py
"""文档系统相关 Pydantic Schema。参照 plans/03_api_interfaces.md §5。"""

from __future__ import annotations

from typing import Any

from pydantic import Field

from app.schemas.common import CamelModel


class DocumentGenerateRequest(CamelModel):
    doc_type: str = Field(description="requirements | design | testing | meeting_minutes | configuration")
    title: str
    context: dict[str, Any] = {}
    template: str | None = None


class DocumentEditRequest(CamelModel):
    content: str | None = None
    title: str | None = None


class DocumentStatusUpdate(CamelModel):
    status: str = Field(description="draft | review | submitted | archived")


class ConversationCreateRequest(CamelModel):
    title: str | None = None
    doc_type: str | None = Field(default=None, description="可选文档类型标签")


class MessageSendRequest(CamelModel):
    instruction: str = Field(description="本轮对用户文档的指令 (新建/扩写/修订等)")


class QaSendRequest(CamelModel):
    instruction: str = Field(description="用户的技术问题")


class ReasoningStep(CamelModel):
    id: str
    title: str
    detail: str | None = None
    status: str = Field(description="pending | running | completed | failed")


class ReasoningSummary(CamelModel):
    status: str = Field(description="thinking | completed | failed")
    steps: list[ReasoningStep] = Field(default_factory=list)


class QaMessage(CamelModel):
    message_id: str
    role: str
    content: str
    reasoning_summary: ReasoningSummary | None = None
    created_at: str


class QaSendResponse(CamelModel):
    assistant_message: QaMessage
    conversation_id: str | None = None


class DocumentSection(CamelModel):
    id: str
    title: str
    status: str = "generated"


class DocumentTaskResponse(CamelModel):
    doc_task_id: str
    status: str
    created_at: Any | None = None
    result_doc_id: str | None = None


class DocumentSummary(CamelModel):
    doc_id: str
    title: str
    type: str
    status: str
    version: str = "V1.0"
    created_at: Any | None = None
    submitted_at: Any | None = None
    generated_by: str | None = None
    file_size: int | None = None


class DocumentDetail(CamelModel):
    doc_id: str
    title: str
    type: str
    version: str = "V1.0"
    status: str
    content: str = ""
    metadata: dict[str, Any] | None = None
    sections: list[dict] = []


class DocumentPreviewResponse(DocumentDetail):
    """文档预览, 字段同详情。"""
